Document Family Center (#307)
discord-userdoccers/discord-userdoccers: pages/resources/family-center.mdx

- **Requestor:** This is the user that _sends_ a link request to a different user and acts as the user the linked user is connected to. Can be viewed as the "parent."
- **Linked user:** This is the user that _receives_ or _accepts_ a link request sent by the requestor and acts as the user the "parent" can view the activity of. Can be viewed as the "teen."
- **Link:** Represents the connection between requestor and linked user.

### Family Center Object

###### Family Center Structure

| Field          | Type                                                                      | Description                                                  |
| -------------- | ------------------------------------------------------------------------- | ------------------------------------------------------------ |
| linked_users   | array[[linked user](#linked-user-object) object]                          | List of linked users                                         |
| teen_audit_log | [teen audit log](#teen-audit-log-object) object                           | Audit log of the linked users activity                       |

> | users          | array[[partial user](/resources/user#partial-user-structure) object]      | List of requestors the linked user is connected to           |

Reviewer: ```suggestion
| users          | array[partial [user](/resources/user#user-object) object]      | List of requestors the linked user is connected to           |
```

  "link_type": 2,
  "link_status": 3,
  "requestor_id": "246877849162743818",
  "user_id": "801318363472330772",
}
```

### Teen Audit Log Object

Audit log of events of the linked user. Visible to both requestors and linked users.

###### Teen Audit Log Structure

| Field               | Type                                                                      | Description                                                                |
| ------------------- | ------------------------------------------------------------------------- | -------------------------------------------------------------------------- |
| teen_user_id        | ?snowflake                                                                | The ID of the linked user                                                  |
| range_start_id      | ?snowflake                                                                | The ID representing the start time of the 7-day track range?               |
| actions             | array[[action](#action-structure) object]                                 | List of [actions](#action-structure) the linked user has done              |

> | users               | array[[partial user](/resources/user#partial-user-structure) object]      | List of users the linked user has interacted with within 7 days            |
> | guilds              | array[[guild](/resources/guild#guild-structure) object]                   | List of guilds the linked user has joined or interacted with within 7 days |

Reviewer: ```suggestion
| users               | array[partial [user](/resources/user#user-object) object]      | Users referenced in the audit log            |
| guilds              | array[[guild](/resources/guild#guild-object) object]                   | Guilds referenced in the audit log |
```

| totals              | object                                                                    | Object keyed by [display types](#display-type), with their values          |

###### Action Structure

| Field           | Type         | Description                                                                                                         |
| --------------- | ------------ | ------------------------------------------------------------------------------------------------------------------- |
| event_id        | snowflake    | The ID of the event action                                                                                          |
| user_id         | snowflake    | The ID of the linked user                                                                                           |

> | entity_id       | snowflake    | The ID of the entity the action relates to (user, guild, or group DM) based off the [display type](#display-type)   |

Reviewer: ```suggestion
| entity_id       | snowflake    | The ID of the entity the action relates to (user, guild, or group DM) based off the [action type](#action-type)   |
```

> | display_type    | integer      | The [display type](#display-type) of the action, detailing what this action involved                                |

Reviewer: ```suggestion
| display_type    | integer      | The [type of the action](#action-type), detailing what this action involved                                |
```

> ###### Display Type

Reviewer: ```suggestion
###### Action Type
```
This enum is named really poorly for some reason

> The Display Type represents what kind of action the linked user engaged in.

Reviewer: ```suggestion
Represents what kind of action the linked user engaged in.
```

| Value     | Description                                                         |
| --------- | ------------------------------------------------------------------- |
| 1         | Users added within the last 7 days                                  |

> | 2         | Server joined within the last 7 days                                |

Reviewer: ```suggestion
| 2         | Guilds joined within the last 7 days                                |
```

| 3         | Users messaged within the last 7 days                               |

> | 4         | Servers the linked user has sent messages in within the last 7 days |

Reviewer: ```suggestion
| 4         | Guilds the linked user has sent messages in within the last 7 days |
```

| 5         | Users called within the last 7 days                                 |

###### Example Teen Audit Log

```json
{
  "teen_user_id": "801318363472330772",
  "range_start_id": "1328607677722394624",
  "actions": [
    {
      "event_id": "1331144363278860318",
      "user_id": "801318363472330772",
      "entity_id": "246877849162743818",
      "display_type": 3
    },
  ],
  "users": [
    {
      "id": "246877849162743818",
      "username": "jay_taelien",
      "global_name": "Jay",
      "avatar": "91b7bc37e924f78625f7ea582fdbac5d",
      "avatar_decoration_data": {
        "asset": "a_aa2e1c2b3cf05b24f6ec7b8b4141f5fc",
        "sku_id": "1144056631374647458",
        "expires_at": null
      },
      "discriminator": "0",
      "public_flags": 16512,

>       "primary_guild": {
>         "identity_guild_id": "1241115476021481582",
>         "identity_enabled": true,
>         "tag": "BUG",
>         "badge": "6cdb13f9bdba1234e711a7c24dce1f51"
>       },
>       "clan": {
>         "identity_guild_id": "1241115476021481582",
>         "identity_enabled": true,
>         "tag": "BUG",
>         "badge": "6cdb13f9bdba1234e711a7c24dce1f51"
>       }

Reviewer: ```suggestion
      "primary_guild": null
```
in the interest of not having to update 3 million examples every time they update the user object, dead fields are omitted and optional structs are usually null

| ----------- | -----------  | ------------------------------------------------------------------------------------------------------------ |
| link_code   | string       | The code used to connect a requestor to a linked user, appended to the end of the URL the QR code encodes    |

<RouteHeader method="GET" url="/users/@me/linked-users">
  Get Linked Users
</RouteHeader>

Returns a [linked users](#linked-users-object) object.

<RouteHeader method="POST" url="/users/@me/linked-users">
  Create Linked User Request
</RouteHeader>

Creates a request that appears in the linked user's Family Center. Returns a [linked users](#linked-users-object) object on success. Fires a [User Update](/topics/gateway-events#user-update) Gateway event.

###### JSON Params

| Field                | Type                               | Description                                                  |
| -------------------- | ---------------------------------- | ------------------------------------------------------------ |
| recipient_id         | snowflake                          | The ID of the user the requestor wants to connect to         |

> | code                 | string                             | The link code from the linked users device                   |

Reviewer: ```suggestion
| code                 | string                             | The link code from the linked user's device                   |
```
